openvm/src/powdr_extension/executor/mod.rs

```diff
@@ -389,6 +390,66 @@ impl PowdrExecutor {
         //         chip.generate_proving_ctx(records)
         //     },
 
+        // merge record arenas, taking care of padding during initialization
+        let mut record_arena_by_air_name = self.record_arena_by_air_name.borrow_mut();
+
+        let mut merged_record_arena_by_air_name = self
+            .record_arena_dimension_by_air_name_per_call
+            .iter()
+            .map(|(air_name, (num_calls, air_width))| {
+                // height is padded to next power of two
+                let merged_record_arena = MatrixRecordArena::with_capacity(
+                    *num_calls * self.number_of_calls(),
+                    *air_width,
+                );
+                (air_name.clone(), merged_record_arena)
+            })
+            .collect::<HashMap<String, MatrixRecordArena<BabyBear>>>();
+
+        for per_call in record_arena_by_air_name.iter_mut() {
+            for (air_name, src_arena) in per_call.drain() {
```

Comment:
So this is where a lot of copying happens? From the single call arenas to the merged arena?